readme_renderer/rst.py

```diff
@@ -129,4 +133,17 @@ def render(
     if rendered:
         return clean(rendered)
     else:
+        # If the warnings stream is empty, docutils had none, so add ours.
+
+        # We have to currently handle the legacy disutils integration check
+        # command, as it implements its own warnings stream structure.
+        # Once that code is removed, we can remove this conditional.
+        if hasattr(stream, "output"):
+            if not stream.output.getvalue():  # type: ignore[union-attr]
+                stream.write("No content rendered from RST source.")
+        elif isinstance(stream, io.StringIO) and not stream.getvalue():
+            stream.write("No content rendered from RST source.")
+        elif isinstance(stream, TextIO) and not stream.readlines():
+            # This is `sys.stderr` from using readme_renderer CLI
+            stream.write("No content rendered from RST source.")
```

Comment:
I'm a little wary of consuming the content on the `TextIO`. Following my suggestion re: `Check`, I think this could be simplified to:

```python
if not stream.read():
    stream.write("No content rendered from RST source.")
stream.seek(0)
```

However it works out, it'd be nice if the `"No content..."` literal was only defined once.